wsd/DocumentBroker.cpp

```diff
@@ -1790,6 +1790,12 @@ void DocumentBroker::handleTileCombinedRequest(TileCombined& tileCombined,
     for (auto& tile : tileCombined.getTiles())
     {
         tile.setVersion(++_tileVersion);
+        if (!hasTileCache())
+        {
+            LOG_WRN("Combined tile request without a loaded document?");
```

Comment:
I suppose we need the loop to setVersion to each tile, otherwise looks like very odd loop considering that hasTileCache() is an invariant in the loop.

But one suggestion is to _move the warning outside the loop_ to do it only once per request, instead of as many times as there are tiles.

I wonder what happens when we try to process these tile requests when the cache is missing. Ideally, that shouldn't result in anything bad, and the tiles get rendered and shown without issue. Let's hope that's the case, but there should be some test, I imagine. Certainly not in scope for this change, though.

Reply:
> But one suggestion is to move the warning outside the loop to do it only once per request, instead of as many times as there are tiles.

Makes sense, will do that in a moment.

> I wonder what happens when we try to process these tile requests when the cache is missing. 

My understanding is that this can only happen when the client intentionally tries to send nonsense commands on the websocket, so the only requirement is that we don't crash on it, but anything else is fine.

> but there should be some test, I imagine. Certainly not in scope for this change, though.

This fix is an exception, because it's a blind fix. I also plan to backport 3ce20bae685e5f345f90c37c5c9c1dcc24fb93b5 from master, and that one will come fix a fix. (The fuzzer can easily just pick up reproducers from a directory, so it's easy to add such "just don't crash" type of tests.)